lib/features/steps/runner_steps.rb

```diff
@@ -64,6 +64,47 @@
   Docker.start_service(service, command: one_line_cmd)
 end
 
+# Allows validation of the last exit code of the last run docker-compose command.
+# Will fail if no commands have been run.
+#
+# @step_input expected_code [Integer] The expected exit code
+Then('the exit code of the last docker command was {int}') do |expected_code|
+  exit_code = Docker.last_exit_code
+  assert_not_nil(exit_code, 'No docker exit code available to verify')
+  assert_equal(exit_code, expected_code)
+end
+
+# A shortcut for the above assuming 0 as a successful exit code
+# Will fail if no commands have been run
+Then('the last run docker command exited successfully') do
+  exit_code = Docker.last_exit_code
+  assert_not_nil(exit_code, 'No docker exit code available to verify')
+  assert_equal(exit_code, 0)
+end
+
+# Allows testing that the last exit code was not 0
+# Will fail if no commands have been run
+Then('the last run docker command did not exit successfully') do
+  exit_code = Docker.last_exit_code
+  assert_not_nil(exit_code, 'No docker exit code available to verify')
+  assert_not_equal(exit_code, 0)
+end
+
+# Allows testing a docker command output a specific string
+# Will fail if no commands have been run
+#
+# @step_input expected_string [String] The string expected in a single log line
+Then('the last run docker command output {string} to stdout') do |expected_string|
```

Comment:
The command is run with `popen2e` so there's no way to know if this was stdout or stderr as they are merged

```suggestion
Then('the last run docker command output {string}') do |expected_string|
```